### PR TITLE
Obstacles

### PR DESCRIPTION
**Pull request by guilbill**
This PR aim to add obstacles to the board. Trello : https://trello.com/c/ndwsjSNg/2-05-as-pierre-i-need-obstacles-because-only-moving-the-puck-on-a-square-board-doesnt-teach-me-a-thing
tipsy-obstacles

### CODE REVIEW

File: src/board.py [hunk 1]
```diff
@@ -28,27 +35,15 @@ def __initialize_empty_board(self):
...
-        pucks = [node for node, attributes in self.graph.nodes(data=True) if attributes.get('puck')]
+        pucks = [node for node, attributes in self.graph.nodes(
+            data=True) if attributes.get('puck')]
         for puck in pucks:
             self.__move_puck_to(puck, direction)
 
     def __get_next_free_node(self, node, direction):
         next_node = [next_node for start_node, next_node, edge_attrs in self.graph.out_edges(
-            node, data=True) if edge_attrs.get('direction') == direction]
+            node, data=True) if (edge_attrs.get('direction') == direction and not self.graph.nodes[next_node].get('obstacle'))]
```
Comment (fzaninotto) [inline src/board.py]: you should remove the edges to obstacles instead

File: src/game.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            self.display_board()
+            self.display_instructions()
+            input_command = ''
+            while (input_command.lower() not in [Board.EAST, Board.WEST, Board.NORTH, Board.SOUTH]):
+                input_command = input('Use ' + Board.NORTH + ', ' + Board.SOUTH +
+                                      ', ' + Board.EAST + ', ' + Board.WEST + ' to tilt the board: ')
+
+            self.board.tilt(input_command)
+
+    def display_board(self):
```
Comment (fzaninotto) [inline src/game.py]: You should unit test this method at least once

File: src/game.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                    line += 'o '
+                elif (self.board.graph.nodes[(i, j)].get('obstacle')):
+                    line += '# '
+                else:
+                    line += '  '
+            line += '#'
+            print(line)
+        print('# '*(self.board.WIDTH+2))
+
+    def display_title(self):
```
Comment (fzaninotto) [inline src/game.py:40]: 👍 badass

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: src/game.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        print("  888    888   888   888 `\"Y88b.    `88..8'")
+        print("  888 .  888   888   888 o.  )88b    `888'")
+        print("  \"888\" o888o  888bod8P' 8\"\"888P'     .8'")
+        print("               888                .o..P'")
+        print("              o888o               `Y8P'")
+        print("")
+        print("")
+        input("Welcome to Tipsy game. Type Enter to start")
+
+    def display_instructions(self):
```
Comment (zyhou) [inline src/game.py:54]: :+1:

File: src/board.py (hunk 1 above)
Comment (guilbill) [reply src/board.py]: done

File: src/game.py (hunk 2 above)
Comment (guilbill) [reply src/game.py]: You mean unit test the display of the board?

File: src/board.py [hunk 5]
```diff
@@ -18,8 +18,7 @@ def __initialize_pucks_positions(self):
         self.graph.nodes[(self.WIDTH//2, self.HEIGHT//2)]["puck"] = 'O'
 
     def __initialize_obstacles_positions(self, obstacles):
-        for obstacle in obstacles:
-            self.graph.nodes[obstacle]["obstacle"] = True
+        self.graph.remove_nodes_from(obstacles)
```
Comment (zyhou) [inline src/board.py:21]: Nice update

File: src/game.py (hunk 2 above)
Comment (fzaninotto) [reply src/game.py]: yes. That will force you to separate string formatting and string output.

File: src/board.py (hunk 1 above)
Comment (fzaninotto) [reply src/board.py]: should be removed

Comment (guilbill) [reply src/board.py]: This time it's really gone

File: src/game.py (hunk 2 above)
Comment (guilbill) [reply src/game.py]: done

**Review by fzaninotto — APPROVED** [review]
Perfect!

**Pull request merged**